src/cpp/src/llm_pipeline.cpp

```diff
@@ -96,6 +96,10 @@ class StatefulLLMPipeline final : public LLMPipelineImplBase {
     ) override {
         auto start_time = std::chrono::steady_clock::now();
         GenerationConfig config = (generation_config.has_value()) ? *generation_config : m_generation_config;
+        // If eos_token_id was not provided, take value from default m_generation_config
```

Comment:
it's overridden in overload working with `EncodedInputs`:
https://github.com/openvinotoolkit/openvino.genai/blob/be083bba756fa53c6ace0264fd92c6f4d047de50/src/cpp/src/llm_pipeline.cpp#L200-L203

Reply:
we can actually drop `GenerationConfig config = (generation_config.has_value()) ? *generation_config : m_generation_config;` and bypass `generation_config` as is